criu/mem.c

```diff
@@ -131,6 +140,171 @@ int init_pidfd_store_sk(pid_t pid, int sk)
 	return -1;
 }
 
+void free_pidfd_store(void)
+{
+	int i;
+	struct pidfd_entry *entry;
+	struct hlist_node *tmp;
+
+	for (i = 0; i < PIDFD_HASH_SIZE; i++) {
+		hlist_for_each_entry_safe(entry, tmp, &pidfd_hash[i], hash) {
+			close(entry->pidfd);
+			xfree(entry);
+		}
+	}
+
+	close_safe(&pidfd_store_sk);
+}
+
+int init_pidfd_store_hash(void)
+{
+	int i, cnt, ret;
+	struct pidfd_entry *entry;
+
+	for (i = 0; i < PIDFD_HASH_SIZE; i++)
+		INIT_HLIST_HEAD(&pidfd_hash[i]);
+
+	/* Skip building pidfd_hash if pidfd_store_sk is not initialized */
+	if (pidfd_store_sk == -1)
+		return 0;
+
+	/*
+	 * Drain all queued pidfd entries in pidfd_store_sk from
+	 * the last predump into pidfd_hash.
+	 */
+	cnt = 0;
+	while (1) {
+		entry = xmalloc(sizeof(struct pidfd_entry));
```

Comment:
if (entry == NULL) ...

Reply:
I don't understand what is the problem, could you please clarify?

Reply:
@ZeyadYasser you should check error on allocation.

Reply:
e.g., something like this
```C
entry = xmalloc(sizeof(struct pidfd_entry));
if (entry == NULL) {
    pr_perror("Can't allocate entry");
    return -1;
}